src/validation/statement.rs

```diff
@@ -531,21 +531,28 @@ fn validate_assignment(
     context: &ValidationContext,
 ) {
     if let Some(left) = left {
-        // check if we assign to a constant variable
+        // Check if we are assigning to a...
         if let Some(StatementAnnotation::Variable { constant, qualified_name, .. }) =
             context.annotations.get(left)
         {
+            // ...constant variable
             if *constant {
                 validator.push_diagnostic(Diagnostic::cannot_assign_to_constant(
                     qualified_name.as_str(),
                     left.get_location(),
                 ));
             }
+
+            // ...VAR_INPUT {ref} variable
```

Comment:
the StatementAnnotation::Variable already carries part of this information in its variableType:
https://github.com/PLC-lang/rusty/blob/98ceaca8b3ecff9e8d5a63f1ab5c808466f3acd5/src/resolver.rs#L165

maybe you can improve the field to also carry the byref/byval information? then you could use the annotation that's already here and we can do this validation without additional costs.

Reply:
Done, I've replaced that field with `ArgumentType` and added some util functions such as `get_inner_ref` to get the wrapped `VariableType`.